### PR TITLE
Pi-Blog app - Add forms

### PR DESCRIPTION
**Pull request by Eccb7**
In the PR, I have adhered to the following project requirements: 

-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Eccb7 ,






Great so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Nice project.✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._

- [ ] the pages are still blank even after when i created some users in the rails console, if this is an oversight from me, please, reach out. however, the pages are blank.




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      @comments = @post.comments.order(created_at: :desc)
+      @post.update(comments_counter: @comments.count)
+
+      # Redirect or render as needed
+      redirect_to user_post_path(@user, @post), notice: 'Comment added successfully'
+    else
+      # Handle validation errors
+      render 'posts/show'
+    end
+  end
```
Comment (Strangeal) [inline app/controllers/comments_controller.rb]: - [ ] Great job so far in your implementation of the forms on your blog app. However, I noticed a few issues that when adjusted would help improve the user experience and functionalities of the app.
   - I noticed you have your comment form placed in the post details page which looks okay but doesn't match project requirements. Your comment form should have a dedicated route(`/posts/:post_id/comments/new`) just like you did for the post form.
   
   - Also, when creating a comment it throws an error that prevents users from commenting on posts. Below is a full details of the error in a screenshot.
   
Screenshot 2024-01-20 at 21 59 27

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1,25 +1,55 @@
...
-      <label>
-        Likes: <span><%= @post.likes_counter %></span>
-      </label>
+    <div class="post-metadata row text-end">
+      <div class="col-md-6">
+        <label class="me-2">Comments: <span id="comments_counter"><%= @post.comments_counter %></span></label>
+      </div>
+      <div class="col-md-6">
+        <label>Likes: <span><%= @post.likes_counter %></span></label>
+      </div>
```
Comment (Strangeal) [inline app/views/posts/show.html.erb]: Great job so far in your implementation of the forms on your blog app. However, I noticed a few issues that when adjusted would help improve the user experience and functionalities of the app.
  -  Kindly consider implementing the like functionality such that when a user click on the like button a like gets created.

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @Eccb7, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Post form 🚀✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Eccb7** [comment]
:pray:

File: app/views/posts/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <%= form.label :text, class: "form-label" %>
+    <%= form.text_area :text, class: "form-control" %>
+  </div>
+
+  <div class="text-end">
+    <%= form.submit 'Create Post', class: "btn btn-primary" %>
+  </div>
+<% end %>
+
+<script>
```
Comment (Olamarx) [inline app/views/posts/new.html.erb:19]: - [ ] I left a detailed issue in the video link bebolw, kindly work on the subbmission of the post, without that, I cant confirm the comment and the likes
[the link](https://www.loom.com/share/b962a1f15ccf41a98755783d55d48b3c?sid=0074944e-3279-4898-9f6d-5cf84f8f731f)

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Eccb7, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Project renders ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**